libs/isograph-react/src/loadable-hooks/useConnectionSpecPagination.ts

```diff
@@ -81,17 +73,26 @@ type NonNullConnection<T> = {
   readonly pageInfo: PageInfo;
 };
 
+type UseConnectionSpecPaginationArgs = {
+  first: number;
+  after: string | null;
+};
+
 export function useConnectionSpecPagination<
   TReadFromStore extends {
-    parameters: {
-      readonly first?: number | void | null;
-      readonly after?: string | void | null;
-    };
+    parameters: object;
     data: object;
   },
   TItem,
 >(
-  loadableField: LoadableField<TReadFromStore, Connection<TItem>>,
+  loadableField: LoadableField<
+    TReadFromStore,
+    Connection<TItem>,
+    UseConnectionSpecPaginationArgs
+  >,
+  initialArgs?: {
+    after?: string | void | null;
```

Comment:
let's add `hasNextPage` here as well. Also, is `initialArgs?: { after: string } | null` a better type for this?

Reply:
Then user has to pass `firstPage.pageInfo.endCursor && { after: firstPage.pageInfo.endCursor }`

Reply:
I think it's up to you to decide
- a lot of users will want an initial page, so it makes sens to take `pageInfo`,
- tbh. I like what we have now the most, I think it's a good low level api, and to me the argument `after` is more readable then `endCusor`,
- overall it shouldn't matter much what we go for, as long as user can make a custom hook around this, then he can even take an array of initial connections and merge those.

Reply:
yeah, I think what draws me to passing `pageInfo` is:
- that is the type of the hooks internal state (in the same way that skip is the state of the use skip limit hook. Skip also happens to be the parameter there, but I'd make the case that it would make sense to also add `hasNextPage` to the state there, corresponding to whether the last request was empty.)
- you can just pass in the `pageInfo` directly, so you don't have to actually type out `endCursor` and `hasNextPage`.
- the hook is lying when it says `hasNextPage: true` if we know that the previous page info had `hasNextPage: false` :/

Reply:
In this specific case, it would also work to take the entire connection. Then, the user wouldn't have to concatenate. But that only works because the return value includes the page info, which is the state. In the skip/limit pagination case, we don't return the `fetchedSoFar` field, so that's not possible :/

Reply:
Do we want `Optional<PageInfo>`?

Reply:
What would the use case be for that?